### ISSUE TITLE
Leader Election Lost

### ISSUE THREAD

**Issue by avwede** [ansible/tower-operator#808]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
The awx-operator-controller-manager pod is giving an error for the awx-manager container, saying that the leader election is lost.

##### ENVIRONMENT
* AWX version: 0.17.0
* Operator version: 0.17.0
* Kubernetes version: v1.22.6+k3s1
* AWX install method: k3s on localhost AlmaLinux

##### STEPS TO REPRODUCE
Clone awx-operator github [https://github.com/ansible/awx-operator#basic-install-on-minikube-beginner-or-testing] and run make deploy.

##### EXPECTED RESULTS

Successfully running awx-operator pod.

##### ACTUAL RESULTS

awx-operator pod in error state with 1/2 containers ready.

##### AWX-OPERATOR LOGS
E0228 16:16:23.993800       7 leaderelection.go:325] error retrieving resource lock awx-operator/awx-operator: Get "https://10.43.0.1:443/api/v1/namespaces/awx-operator/configmaps/awx-operator": context deadline exceeded
{"level":"error","ts":1646064984.1964557,"logger":"cmd","msg":"Proxy or operator exited with error.","Namespace":"awx-operator","error":"leader election lost","stacktrace":"github.com/operator-framework/operator-sdk/internal/cmd/ansible-operator/run.NewCmd.func1\n\t/workspace/internal/cmd/ansible-operator/run/cmd.go:79\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897\nmain.main\n\t/workspace/cmd/ansible-operator/main.go:40\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"}

**Comment by kaushik4r** [comment]
What was the fix for the issue? I still come across this error in awx-operator 0.18.0 -

`I0408 20:50:43.021280       7 leaderelection.go:253] successfully acquired lease ansible-awx/awx-operator
10
{"level":"info","ts":1649451043.0215724,"logger":"controller-runtime.manager.controller.awxrestore-controller","msg":"Starting EventSource","source":"kind source: awx.ansible.com/v1beta1, Kind=AWXRestore"}
9
{"level":"info","ts":1649451043.0216498,"logger":"controller-runtime.manager.controller.awxrestore-controller","msg":"Starting Controller"}
8
{"level":"info","ts":1649451043.0216103,"logger":"controller-runtime.manager.controller.awx-controller","msg":"Starting EventSource","source":"kind source: awx.ansible.com/v1beta1, Kind=AWX"}
7
{"level":"info","ts":1649451043.0216656,"logger":"controller-runtime.manager.controller.awx-controller","msg":"Starting Controller"}
6
{"level":"info","ts":1649451043.0218334,"logger":"controller-runtime.manager.controller.awxbackup-controller","msg":"Starting EventSource","source":"kind source: awx.ansible.com/v1beta1, Kind=AWXBackup"}
5
{"level":"info","ts":1649451043.0219219,"logger":"controller-runtime.manager.controller.awxbackup-controller","msg":"Starting Controller"}
4
{"level":"info","ts":1649451043.1225674,"logger":"controller-runtime.manager.controller.awxrestore-controller","msg":"Starting workers","worker count":64}
3
{"level":"info","ts":1649451043.1226282,"logger":"controller-runtime.manager.controller.awx-controller","msg":"Starting workers","worker count":64}
2
{"level":"info","ts":1649451043.122675,"logger":"controller-runtime.manager.controller.awxbackup-controller","msg":"Starting workers","worker count":64}
1
E0411 12:10:02.846570       7 leaderelection.go:325] error retrieving resource lock ansible-awx/awx-operator: etcdserver: request timed out`

**Comment by estsauver** [comment]
I'm still seeing this as well

**Comment by Sennar19** [comment]
Hi guys!

Same error for me, these are my logs:

```
{"level":"info","ts":"2023-07-07T10:19:40Z","logger":"proxy","msg":"Starting to serve","Address":"127.0.0.1:8888"}
{"level":"info","ts":"2023-07-07T10:19:40Z","logger":"apiserver","msg":"Starting to serve metrics listener","Address":"localhost:5050"}
{"level":"info","ts":"2023-07-07T10:19:40Z","msg":"Starting server","path":"/metrics","kind":"metrics","addr":"127.0.0.1:8080"}
{"level":"info","ts":"2023-07-07T10:19:40Z","msg":"Starting server","kind":"health probe","addr":"[::]:6789"}
I0707 10:19:40.460565       7 leaderelection.go:248] attempting to acquire leader lease awx/awx-operator...
E0707 10:20:06.312752       7 leaderelection.go:367] Failed to update lock: etcdserver: request timed out
I0707 10:20:10.211452       7 leaderelection.go:258] successfully acquired lease awx/awx-operator
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting EventSource","controller":"awx-controller","source":"kind source: *unstructured.Unstructured"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting Controller","controller":"awx-controller"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting EventSource","controller":"awxrestore-controller","source":"kind source: *unstructured.Unstructured"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting Controller","controller":"awxrestore-controller"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting EventSource","controller":"awxbackup-controller","source":"kind source: *unstructured.Unstructured"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting Controller","controller":"awxbackup-controller"}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting workers","controller":"awxbackup-controller","worker count":3}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting workers","controller":"awx-controller","worker count":3}
{"level":"info","ts":"2023-07-07T10:20:10Z","msg":"Starting workers","controller":"awxrestore-controller","worker count":3}
E0707 10:20:42.469295       7 leaderelection.go:367] Failed to update lock: etcdserver: request timed out
I0707 10:20:42.855247       7 leaderelection.go:283] failed to renew lease awx/awx-operator: timed out waiting for the condition
{"level":"error","ts":"2023-07-07T10:20:42Z","logger":"cmd","msg":"Proxy or operator exited with error.","Namespace":"awx","error":"leader election lost","stacktrace":"github.com/operator-framework/operator-sdk/internal/cmd/ansible-operator/run.run\n\t/workspace/internal/cmd/ansible-operator/run/cmd.go:287\ngithub.com/operator-framework/operator-sdk/internal/cmd/ansible-operator/run.NewCmd.func1\n\t/workspace/internal/cmd/ansible-operator/run/cmd.go:80\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968\nmain.main\n\t/workspace/cmd/ansible-operator/main.go:40\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
```

**Comment by pachumon** [comment]
i am also seeing this issue in 2.7.0 release any inputs would be really helpful

**Comment by orenstartio** [comment]
@avwede Can you explain why did you close this issue? I'm having this issue as well on version `2.9.0`.
Can you explain how you fixed it? And if not, can you reopen it please?

**Comment by ibrahimkk-moideen** [comment]
@avwede Seeing the same issue. Can you post the fix here.

**Comment by gstefanoff** [comment]
Seeing the same issue, where can the fix be found? Running operator version 2.19.1

`I0411 11:00:12.970196       7 leaderelection.go:285] failed to renew lease awx/awx-operator: timed out waiting for the condition
{"level":"error","ts":"2025-04-11T11:00:12Z","logger":"cmd","msg":"Proxy or operator exited with error.","error":"leader election lost","stacktrace":"github.com/operator-framework/ansible-operator-plugins/internal/cmd/ansible-operator/run.run\n\tansible-operator-plugins/internal/cmd/ansible-operator/run/cmd.go:261\ngithub.com/operator-framework/ansible-operator-plugins/internal/cmd/ansible-operator/run.NewCmd.func1\n\tansible-operator-plugins/internal/cmd/ansible-operator/run/cmd.go:81\ngithub.com/spf13/cobra.(*Command).execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115\ngithub.com/spf13/cobra.(*Command).Execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039\nmain.main\n\tansible-operator-plugins/cmd/ansible-operator/main.go:40\nruntime.main\n\t/opt/hostedtoolcache/go/1.20.12/x64/src/runtime/proc.go:250"}`

**Comment by rkotech** [comment]
This issue seems to be closed, even though there are no solutions here.

I have the same issue in 2026, more than 4 years after the issue was opened.
Can anyone sched some light on this please?
